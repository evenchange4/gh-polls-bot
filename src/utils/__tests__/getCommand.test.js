// @flow
const getCommand = require('../getCommand');

it('should return correct regular expression', () => {
  expect(getCommand(`/polls`)).toEqual([]);
  expect(getCommand(`/polls `)).toEqual(['/polls ', '']);
  expect(getCommand(`/polls 1`)).toEqual(['/polls 1', '1']);
  expect(getCommand(`/Polls 1`)).toEqual([]);
  expect(getCommand(`/pollss 1`)).toEqual([]);
  expect(getCommand(`/polls 1 2`)).toEqual(['/polls 1 2', '1 2']);
  expect(getCommand(`/polls 1 2  3`)).toEqual(['/polls 1 2  3', '1 2  3']);
  expect(getCommand(`/polls 1 2  3 4 `)).toEqual([
    '/polls 1 2  3 4 ',
    '1 2  3 4 ',
  ]);
  expect(
    getCommand(`/polls 1 2  3 4
5`),
  ).toEqual(['/polls 1 2  3 4', '1 2  3 4']);
  expect(getCommand(`/polls 1 '23' "5 5" 66`)).toEqual([
    `/polls 1 '23' "5 5" 66`,
    `1 '23' "5 5" 66`,
  ]);
  expect(getCommand(`/polls option1 'option 2' "option 3" `)).toEqual([
    `/polls option1 'option 2' "option 3" `,
    `option1 'option 2' "option 3" `,
  ]);
  expect(getCommand(`/polls/travis/.rvm/rubies`)).toEqual([]);
  expect(getCommand(`/polls  1  2  3  `)).toEqual([
    `/polls  1  2  3  `,
    `1  2  3  `,
  ]);
  expect(getCommand(`/polls\t1\t2\t3\ttab`)).toEqual([
    `/polls\t1\t2\t3\ttab`,
    `1\t2\t3\ttab`,
  ]);
  expect(
    getCommand(
      `/home/travis/.rvm/rubies/ruby-2.4.0/lib/ruby/site_ruby/2.4.0/bundler/templates/Executable`,
    ),
  ).toEqual([]);
});

it('should return command with body content', () => {
  const body = `
# H1
## H2
> code
/polls 1 2 3
  4`;
  expect(getCommand(body)).toEqual(['/polls 1 2 3', '1 2 3']);
});

it('should return only one command', () => {
  const body = `
# H1
## H2
> code
/polls 4 5 6
/polls 1 2 3
  `;
  expect(getCommand(body)).toEqual(['/polls 4 5 6', '4 5 6']);
});

it('should return the first matched command', () => {
  const body = `
# H1
## H2
/polls 7 8 69
> code
/polls 1 2 3
  `;
  expect(getCommand(body)).toEqual(['/polls 7 8 69', '7 8 69']);
});

it('should return command without argument', () => {
  const body = `
# H1
## H2
/polls
  `;
  expect(getCommand(body)).toEqual([]);
});

it('should return empty array without any argument matched issue#13', () => {
  const body = `Errno::ENOENT: No such file or directory @ rb_sysopen -
/home/travis/.rvm/rubies/ruby-2.4.0/lib/ruby/site_ruby/2.4.0/bundler/templates/Executable
An error occurred while installing concurrent-ruby (1.0.5), and Bundler cannot
continue.
Make sure that gem install concurrent-ruby -v '1.0.5' succeeds before
bundling.`;
  expect(getCommand(body)).toEqual([]);
});
