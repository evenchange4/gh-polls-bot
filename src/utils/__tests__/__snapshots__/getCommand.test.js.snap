// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return command with body content 1`] = `
Array [
  "/polls 1 2 3",
  "1 2 3",
]
`;

exports[`should return command without argument 1`] = `
Array [
  "/polls",
  undefined,
]
`;

exports[`should return only one command 1`] = `
Array [
  "/polls 4 5 6",
  "4 5 6",
]
`;

exports[`should return the first matched command 1`] = `
Array [
  "/polls 7 8 69",
  "7 8 69",
]
`;
